tests/containers/podmansh.pm

```diff
@@ -0,0 +1,200 @@
+# SUSE's openQA tests
+#
+# Copyright 2021-2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Test rootless mode on podman.
+# - add a user on the /etc/subuid and /etc/subgid to allow automatically allocation subuid and subgid ranges.
+# - check uids allocated to user (inside the container are mapped on the host)
+# - give read access to the SUSE Customer Center credentials to call zypper from in the container.
+#   This grants the current user the required access rights
+# - Test rootless container:
+#   * container is launched with default root user
+#   * container is launched with existing user id
+#   * container is launched with keep-id of the user who run the container
+# - Restore /etc/zypp/credentials.d/ credentials
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use Utils::Systemd qw(systemctl);
+use utils;
+use containers::common;
+
+my $quadlet_dir = '/etc/containers/systemd';
+
+my $src_image = "registry.opensuse.org/opensuse/leap:15.6";
+my $quadlet_container = <<_EOF_;
+[Unit]
+Description=The podmansh container
+After=local-fs.target
+
+[Container]
+Image=$src_image
+ContainerName=podmansh
+UserNS=keep-id
+RunInit=yes
+DropCapability=all
+NoNewPrivileges=true
+
+# change homedir to match 'user-console' needle
+Environment=HOME=/home
+WorkingDir=/home
+
+# OpenQA serial terminal access
+AddDevice=/dev/ttyS0
+
+Exec=sleep infinity
+
+[Quadlet]
+# avoid infinite waiting for network by podman-user-wait-network-online.service
+# https://github.com/systemd/systemd/issues/28762
+DefaultDependencies=false
+
+[Install]
+RequiredBy=default.target
+_EOF_
+
+my $unit_name = 'podmansh.container';
+my $user;
```

Comment:
This variable is essentially only `$testapi::username` and should IMHO not exist on a module level. Perhaps it';s better to either replace it with `$testapi::username` or move it to the functions themselves, if we want to use shorter names?